File: android_apps/tensorflow_for_poets_tflite/app/src/main/java/scanfit/modeling/MuscleGroup.java
package scanfit.modeling;

import java.io.Serializable;

public class MuscleGroup implements Serializable {
    public String name;

    public MuscleGroup(String name) {
        this.name = name;
    }

    public boolean equals(MuscleGroup other) {
        return this.name.equalsIgnoreCase(other.name);
    }

    @Override
    public boolean equals(Object other) {
        try {
            return ((MuscleGroup) other).equals(this);
        } catch (ClassCastException e) {
            return other.equals(this);
        }
    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }

    @Override
    public String toString() {
        return name;
    }
}

File: android_apps/tensorflow_for_poets_tflite/app/src/main/java/com/scanfit/modeling/MuscleGroup.kt
package com.scanfit.modeling

import java.io.Serializable

class MuscleGroup(var name: String) : Serializable {

    fun equals(other: MuscleGroup): Boolean {
        return this.name.equals(other.name)
    }

    override fun equals(other: Any?): Boolean {
        try {
            return (other as MuscleGroup).equals(this)
        } catch (e: ClassCastException) {
            return other == this
        }

    }

    override fun hashCode(): Int {
        return name.hashCode()
    }

    override fun toString(): String {
        return name
    }
}
